var/spack/repos/builtin/packages/relion/package.py

```diff
@@ -15,10 +15,18 @@ class Relion(CMakePackage, CudaPackage):
     homepage = "http://http://www2.mrc-lmb.cam.ac.uk/relion"
     git      = "https://github.com/3dem/relion.git"
 
+    # 3.1 is beta but referenced in published papers
+    # prefer stable 3.0 until no longer beta
+    version('3.1_beta', branch='ver3.1')
+
+    # 3.0.8 latest release in 3.0 branch
+    # prefer for now
+    version('3.0.8', tag='3.0.8', preferred=True)
     version('3.0.7', tag='3.0.7')
-    # relion has no develop branch though pulling from master
-    # should be considered the same as develop
-    version('develop', branch='master')
+
+    # relion master contains development code
+    # contains 3.0 branch code
+    version('master')
```

Comment:
Does this actually work without `branch='master'`?

Reply:
Yes, indeed:

spack install relion@master %gcc@7.4.0 cuda_arch=60 ^cuda ^openmpi@4.0.1+cuda+pmi+legacylaunchers schedulers=slurm fabrics=verbs 

```
==> pkgconf is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/pkgconf-1.6.3-6pbd3woum33nlydzl6wtjtkr5oj4qofn
==> ncurses is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/ncurses-6.1-d62oojnju2ue5ymqiyje7h7qgm7au5qh
==> openssl@1 : externally installed in /usr
==> openssl@1 : already registered in DB
==> cmake is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/cmake-3.15.4-ozxjsikns4ixmajgoq7ft7g7p4ijc5mg
==> cuda is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/cuda-10.1.243-x5r677g66r6yrkdmcxq75up2pab2ttw7
==> libsigsegv is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libsigsegv-2.12-5kxfqjxgvxep5f4cu34twuice6p5lkzc
==> m4 is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/m4-1.4.18-zdfqe5ib5ogi6z2wmimyokfsscgwoxyt
==> libtool is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libtool-2.4.6-jkaonknwdlev3q6hanpkb3pzft7gnuak
==> util-macros is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/util-macros-1.19.1-zfflqeytfwt5wzwjtjpte426hqhuedrd
==> libpciaccess is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libpciaccess-0.13.5-zzkt7ck3y5qfl3bi73hqk7a6avjn4rap
==> libiconv is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libiconv-1.16-meyg2avc5ldglxqayvrmzltjirkpgplv
==> xz is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/xz-5.2.4-gqw2wk5johsjx3cxit4uficfuggdbeqi
==> zlib is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/zlib-1.2.11-w4v4ex6wf3qcmkuxdnmga7snctdxiz6h
==> libxml2 is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libxml2-2.9.9-3qwzb7r6foevl6dtibgteedru4rhuzdc
==> readline is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/readline-8.0-tdxl2p5g2vdc73o6ua5xrhg7zzztrcwk
==> gdbm is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/gdbm-1.18.1-cojsnrtd7r2k2epafo63zpwxsq373qnl
==> perl is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/perl-5.30.0-tfhrcw7tv65ix4c2gyfxq5zp7vnfu7ib
==> autoconf is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/autoconf-2.69-jfbpnueijtxtx6i4eiqljieqxazj65cv
==> automake is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/automake-1.16.1-ccxyrpcrukfakguo5vqxfe3arm3f4dou
==> numactl is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/numactl-2.0.12-s2vakqwjducrcfgkzws2khzeljcq5wxi
==> hwloc is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/hwloc-1.11.11-evcj3wofwwelx7wwttegpl25a22divvm
==> slurm@17 : externally installed in /usr
==> slurm@17 : already registered in DB
==> openmpi is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/openmpi-4.0.1-f47m4fwwkfy7ccuhc6vsh3jehm4ybufz
==> fftw is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/fftw-3.3.8-6nrhy452cjxivqy6izjjz4mbaxd6hlsk
==> inputproto is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/inputproto-2.3.2-4zm4whuu7dvt7y3n7cqfzriyadm65onj
==> kbproto is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/kbproto-1.0.7-h7okgzts3voxoa5szbdqfebkptey3qwx
==> libpthread-stubs is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libpthread-stubs-0.4-mcmitieengwwro4diya2p7rc2ylqdq6j
==> xproto is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/xproto-7.0.31-ihhs3gmu3a7wbp2kgecceoz7vsioaa6y
==> libxau is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libxau-1.0.8-xyz4rl2nohcrdufgo6zlw6sfwmlboepn
==> libbsd is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libbsd-0.9.1-5mnn25dshcm2pl7ypkfpzszgpdzkwjfv
==> libxdmcp is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libxdmcp-1.1.2-alyknxzezksv75ps7d2rj5mdtuq6lztz
==> xcb-proto is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/xcb-proto-1.13-qzufcxddcltqiwzym5ozfxjgeuse6qtu
==> libxcb is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libxcb-1.13-ez7koil452pl5mthewx3f6an3rz54aun
==> xextproto is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/xextproto-7.3.0-57gsbkc2lbgzxnym5chc4u7kbgwy6fg6
==> xtrans is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/xtrans-1.3.5-ymep6weasm2vlm3taly3wbfoerg7azsc
==> libx11 is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libx11-1.6.7-vfw5rbkzzwbhmgvm6ddgyloodijlu7yf
==> fltk is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/fltk-1.3.3-7j7ovvocqxn3ea5o7kw6i75bltxv2a42
==> nasm is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/nasm-2.14.02-ayrtc5mp7rnwagprd72arkztup6jso3n
==> libjpeg-turbo is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libjpeg-turbo-2.0.3-2ebud7veklupbe42rjiyk45lgw2bedsv
==> libtiff is already installed in /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/libtiff-4.0.10-hckkioka2tvtjjxkfpb5k5h2snl24yj6
==> Installing relion
==> Searching for binary cache of relion
==> Warning: No Spack mirrors are currently configured
==> No binary for relion found: installing from source
==> Cloning git repository: https://github.com/3dem/relion.git
==> No checksum needed when fetching with git
==> Already staged spack-stage-relion-master-ebke3cdc3whtp2u3nt77trc5dhgmsgya in /tmp/dod2014/spack-stage/spack-stage-relion-master-ebke3cdc3whtp2u3nt77trc5dhgmsgya
==> No patches needed for relion
==> Building relion [CMakePackage]
==> Executing phase: 'cmake'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed relion
  Fetch: 1.61s.  Build: 3m 52.47s.  Total: 3m 54.08s.
[+] /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/relion-master-ebke3cdc3whtp2u3nt77trc5dhgmsgya
```

Present:

```
==> Executing phase: 'install'

==> Successfully installed relion
  Fetch: 1.61s.  Build: 3m 52.47s.  Total: 3m 54.08s.
[+] /pbtech_mounts/homes027/dod2014/prepos/spack/opt/spack/linux-centos7-broadwell/gcc-7.4.0/relion-master-ebke3cdc3whtp2u3nt77trc5dhgmsgya
```

3.0.8 compiled from master:

```
dod2014@node175[~/prepos/spack]$ spack load -r /ebke3cd

dod2014@node175[~/prepos/spack]$ relion_refine --version
RELION version: 3.0.8 
Precision: BASE=double, CUDA-ACC=single 
```